### PR TITLE
Add login capabilities to the OCI client

### PR DESCRIPTION
**Pull request by adrien-f**
Hi 👋 

This can be re-used by flux to auto-login during artifacts operations as showcased in https://github.com/fluxcd/flux2/pull/3079

Let me know if you have any input!

Signed-off-by: Adrien Fillon <adrien.fillon@manomano.com>

### CODE REVIEW

**Comment by adrien-f** [comment]
I'm not 100% happy with the tests as I had to dig deep to verify the behavior as options are private at some point. Let me know if you have some input 😄

File: oci/client/login.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	if err != nil {
+		return fmt.Errorf("could not login to provider %v with url %s: %w", provider, url, err)
+	}
+
+	c.options = append(c.options, crane.WithAuth(authenticator))
+	return nil
+}
+
+// AutoLogin configures the client to autologin with current supported providers based on the URL
+func (c *Client) AutoLogin(ctx context.Context, mgr *login.Manager, url string) error {
```
Comment (stefanprodan) [inline oci/client/login.go:70]: We don't need this anymore right?

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @adrien-f 🏅 

PS. Please rebase with upstream main and force push so we can merge this.

**Comment by adrien-f** [comment]
> LGTM
> 
> Thanks @adrien-f 🏅
> 
> PS. Please rebase with upstream main and force push so we can merge this.

Amazing 🙌 It's rebased and pushed :shipit:

File: oci/client/login.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package client
```
Comment (stefanprodan) [inline oci/client/login.go:1]: Please add the Copyright heading

**Pull request merged**